File: app/src/main/java/tech/blur/bookcrossing/features/core/GlobalNavigator.java
package tech.blur.bookcrossing.features.core;

import android.app.Activity;
import android.content.Intent;

import ru.terrakok.cicerone.Navigator;
import ru.terrakok.cicerone.commands.Command;
import ru.terrakok.cicerone.commands.Forward;
import tech.blur.bookcrossing.features.books.bookshelf.BookShelfActivity;

public class GlobalNavigator implements Navigator {

    private Activity activity;

    public GlobalNavigator(Activity activity) {
        this.activity = activity;
    }

    @Override
    public void applyCommands(Command[] commands) {
        if (commands[0] instanceof Forward) {
            Intent intent;
            switch (((Forward) commands[0]).getScreenKey()){
                case Screens.BOOK_SHELF:
                    intent = new Intent(activity, BookShelfActivity.class);
                    activity.startActivity(intent);
                    activity.finish();
                    break;
            }
        }
    }
}

File: app/src/main/java/tech/blur/bookcrossing/features/core/GlobalNavigator.kt
package tech.blur.bookcrossing.features.core

import android.app.Activity
import android.content.Intent

import ru.terrakok.cicerone.Navigator
import ru.terrakok.cicerone.commands.Command
import ru.terrakok.cicerone.commands.Forward
import tech.blur.bookcrossing.features.auth.signin.presentation.SignInActivity
import tech.blur.bookcrossing.features.main.BookShelfActivity

class GlobalNavigator(private val activity: Activity) : Navigator {

    override fun applyCommands(commands: Array<Command>) {
        if (commands[0] is Forward) {
            val intent: Intent
            when ((commands[0] as Forward).screenKey) {
                Screens.BOOK_SHELF -> {
                    intent = Intent(activity, BookShelfActivity::class.java)
                    activity.startActivity(intent)
                    activity.finish()
                }
                Screens.SIGN_IN -> {
                    intent = Intent(activity, SignInActivity::class.java)
                    activity.startActivity(intent)
                    activity.finish()
                }
            }
        }
    }
}
